redoctober_test.go

```diff
@@ -0,0 +1,701 @@
+// Tests the Red October API. These tests ensure consistency in the API's
+// return status codes and data. They do not explicitly test for correctness
+// of the functions called (that is handled by the tests in /core). Running
+// these tests first require that you build this project and have a recent
+// binary of it in either the /redoctober folder or in $GOPATH/bin/redoctober.
+
+package main
+
+import (
+	"bytes"
+	"crypto/tls"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"os/exec"
+	"testing"
+	"time"
+
+	"github.com/cloudflare/redoctober/core"
+)
+
+const baseURL = "https://localhost:8080/"
+
+var (
+	// Inputs that are used in multiple test cases are put here to reduce verbosity.
+	createVaultInput = &core.CreateRequest{Name: "Alice", Password: "Lewis"}
+
+	createUserInput1 = &core.CreateUserRequest{Name: "Bill", Password: "Lizard"}
+	createUserInput2 = &core.CreateUserRequest{Name: "Cat", Password: "Cheshire"}
+	createUserInput3 = &core.CreateUserRequest{Name: "Dodo", Password: "Dodgson"}
+
+	delegateInput1 = &core.DelegateRequest{
+		Name:     createUserInput1.Name,
+		Password: createUserInput1.Password,
+		Time:     "2h34m",
+		Uses:     1,
+	}
+	delegateInput2 = &core.DelegateRequest{
+		Name:     createUserInput2.Name,
+		Password: createUserInput2.Password,
+		Time:     "2h34m",
+		Uses:     1,
+	}
+
+	encryptInput = &core.EncryptRequest{
+		Minimum:  2,
+		Name:     createVaultInput.Name,
+		Password: createVaultInput.Password,
+		Owners:   []string{createUserInput1.Name, createUserInput2.Name},
+		Data:     []byte("V2h5IGlzIGEgcmF2ZW4gbGlrZSBhIHdyaXRpbmcgZGVzaz8K"),
+	}
+)
+
+func init() {
+	// Because the certificate being used is self-signed, InsecureSkipVerify must be enabled
+	// to avoid the POST requests from failing.
+	cfg := &tls.Config{
```

Comment:
Is it not possible to simply inline this declaration?
```
http.DefaultClient.Transport = &http.Transport{
  TLSClientConfig: &tls.Config{
    InsecureSkipVerify: true,
  },
}
```
That would be cleaner, I think.